### PR TITLE
Allow consumers to define tags

### PR DESCRIPTION
**Pull request by pavolloffay**
Allow users to define framework/app... specific tags.

### CODE REVIEW

**Comment by pavolloffay** [comment]
@bensigelman @yurishkuro @michaelsembwever could you please review?

File: opentracing-api/src/main/java/io/opentracing/tag/Tags.java [hunk 1]
```diff
@@ -18,7 +18,8 @@
  * semantic information about the spans. Tracers may expose additional features based on these
  * standardized data points. Tag names follow a general structure of namespacing.
  *
- * @see http://opentracing.io/data-semantics/
+ * @see <a href="http://opentracing.io/documentation/pages/spec.html">http://opentracing.io/documentation/pages/spec
```
Comment (pavolloffay) [inline opentracing-api/src/main/java/io/opentracing/tag/Tags.java]: This was wrong. I'm not sure which link should be used. This one or:

https://github.com/opentracing/specification/blob/master/data_conventions.yaml

File: opentracing-api/src/main/java/io/opentracing/tag/Tags.java [hunk 2]
```diff
@@ -18,7 +18,7 @@
  * semantic information about the spans. Tracers may expose additional features based on these
  * standardized data points. Tag names follow a general structure of namespacing.
  *
- * @see http://opentracing.io/data-semantics/
+ * @see <a href="http://opentracing.io/documentation/pages/spec.html">http://opentracing.io/documentation/pages/spec.html</a>
```
Comment (pavolloffay) [inline opentracing-api/src/main/java/io/opentracing/tag/Tags.java]: This was wrong. I'm not sure which link should be used. This one or:

https://github.com/opentracing/specification/blob/master/data_conventions.yaml

Comment (wu-sheng) [inline opentracing-api/src/main/java/io/opentracing/tag/Tags.java]: I think `https://github.com/opentracing/specification/blob/master/data_conventions.yaml` is better.

Comment (bensigelman) [inline opentracing-api/src/main/java/io/opentracing/tag/Tags.java]: I agree ^^^

**Pull request merged**